mirrord/config/src/lib.rs

```diff
@@ -415,13 +433,14 @@ impl LayerFileConfig {
     where
         P: AsRef<Path>,
     {
-        let path = path.as_ref();
-        let file = std::fs::read(path)?;
-
-        match path.extension().and_then(|os_val| os_val.to_str()) {
-            Some("json") => Ok(serde_json::from_slice::<Self>(&file[..])?),
-            Some("toml") => Ok(toml::from_str::<Self>(&String::from_utf8_lossy(&file))?),
-            Some("yaml") => Ok(serde_yaml::from_slice::<Self>(&file[..])?),
+        let mut template_engine = Tera::default();
+        template_engine.add_template_file(path.as_ref(), Some("main"))?;
+        let rendered = template_engine.render("main", &tera::Context::new())?;
```

Comment:
Maybe add some values like `mirrord-version` & `protocol-version` values to context (`mirrord-version` can be useful if you have a proxy ghcr.io)